app/controllers/purchases_controller.rb

```diff
@@ -3,14 +3,17 @@
 
 class PurchasesController < ApplicationController
 
-  def new
-    item = Item.find(params[:item_id])
-    purchase = Purchase.new
-    @presenter = NewPurchasePresenter.new(purchase, item)
+  before_action :check_current_user
+
+  def check_current_user
```

Comment:
This should be a private method. It may even need to go in ApplicationController or some other shared place dependig on the comment above.